static/app/views/settings/organizationTeams/teamSettings/index.tsx

```diff
@@ -23,44 +24,23 @@ import useApi from 'sentry/utils/useApi';
 import {useNavigate} from 'sentry/utils/useNavigate';
 import useOrganization from 'sentry/utils/useOrganization';
 import {useParams} from 'sentry/utils/useParams';
+import {useTeamsById} from 'sentry/utils/useTeamsById';
 import {ProjectPermissionAlert} from 'sentry/views/settings/project/projectPermissionAlert';
 
-interface TeamSettingsProps {
-  team: Team;
-}
-
-function TeamSettings({team}: TeamSettingsProps) {
-  const params = useParams<{teamId: string}>();
+export default function TeamSettings() {
   const navigate = useNavigate();
   const organization = useOrganization();
   const api = useApi();
 
-  const handleSubmitSuccess: FormProps['onSubmitSuccess'] = (resp: Team, _model, id) => {
-    // Use the old slug when triggering the update so we correctly replace the
-    // previous team in the store
-    updateTeamSuccess(team.slug, resp);
-    if (id === 'slug') {
-      addSuccessMessage(t('Team name changed'));
-      navigate(`/settings/${organization.slug}/teams/${resp.slug}/settings/`, {
-        replace: true,
-      });
-    }
-  };
+  const {teamId} = useParams<{teamId: string}>();
+  const {teams, isLoading} = useTeamsById({slugs: [teamId]});
+  const team = teams.find(tm => tm.slug === teamId);
```

Comment:
i think again we want to pass team down via Outlet context in TeamDetails and it would be a larger refactor since you want to move all the children at the same time

Reply:
Ah okay, I think I understand - going to draft this and run it by you